src/main/kotlin/snyk/container/KubernetesImageCache.kt

```diff
@@ -0,0 +1,87 @@
+package snyk.container
+
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.progress.runBackgroundableTask
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.newvfs.events.VFileContentChangeEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileCopyEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileCreateEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileDeleteEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileMoveEvent
+import com.jetbrains.rd.util.concurrentMapOf
+import org.jetbrains.yaml.YAMLFileType
+
+object KubernetesImageCache {
+    private val images = concurrentMapOf<VirtualFile, Set<KubernetesWorkloadImage>>()
+    internal val logger = Logger.getInstance(javaClass.name)
+
+    fun clear() {
+        images.clear()
+    }
+
+    fun scanProjectForKubernetesFiles(project: Project) {
+        val title = "Snyk: Scanning For Kubernetes Files..."
+        runBackgroundableTask(title, project, true) {
+            ProjectRootManager.getInstance(project).fileIndex.iterateContent {
+                this.extractFromFile(it, project)
+                true
+            }
+        }
+    }
+
+    fun getKubernetesWorkloadFilesFromCache(): Set<VirtualFile> = images.map { entry -> entry.key }.toSet()
```

Comment:
let's use @testOnly for public method used in test only.
also overall it seems not right to expose any internals for test only purposes...

Reply:
It was a requirement of the Jira ticket to provide that method. Supposedly for later tickets. That's why it's there :).